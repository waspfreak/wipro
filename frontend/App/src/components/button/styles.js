import { StyleSheet } from 'react-native';
import colors from '../../styles/colors';
import globalStyles from '../../styles/globalStyles';

export default StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    padding: 10,
    borderRadius: 50,
    flexDirection: 'row',
    minHeight: 20,
    minWidth: 146,
    paddingHorizontal: 32
  },
  content: {
    flexDirection: 'row'
  },
  icon: {
    marginLeft: 5
  },
  containerLarge: {
    width: '100%'
  },
  containerSmall: {
    minWidth: 63,
    paddingHorizontal: 16
  },
  containerDisabled: {
    backgroundColor: colors.lightGrey,
    borderWidth: 0
  },
  containerPrimary: {
    backgroundColor: colors.green,
    borderWidth: 1,
    borderColor: colors.green
  },
  containerSecondary: {
    backgroundColor: colors.white,
    borderWidth: 1,
    borderColor: colors.darkGrey
  },
  containerSecondaryRed: {
    backgroundColor: colors.redOpacity,
    borderWidth: 1,
    borderColor: colors.red
  },
  containerTertiary: {
    backgroundColor: colors.blue,
    borderWidth: 1,
    padding: 5,
    minHeight: 30,
    borderColor: colors.blue,
    minWidth: 60,
    paddingHorizontal: 12
  },
  text: {
    color: colors.white,
    alignSelf: 'center',
    fontSize: 17,
    ...globalStyles.Body
  },
  textDisabled: {
    color: colors.white
  },
  textSecondary: {
    color: colors.black
  },
  textSecondaryRed: {
    color: colors.red
  },
  textTertiary: {
    color: colors.white,
    fontSize: 15
  }
});
