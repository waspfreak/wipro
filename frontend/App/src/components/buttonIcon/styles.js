import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  buttonBackground: {
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 50,
    height: 28,
    width: 28
  }
});
