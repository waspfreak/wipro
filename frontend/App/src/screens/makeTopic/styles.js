import { StyleSheet } from 'react-native';
export default StyleSheet.create({
  container: {},
  mainContent: {
    paddingVertical: 20,
    paddingHorizontal: 20
  },
  header: {
    paddingHorizontal: 16
  },
  input: {
    width: '100%',
    height: 80,
    marginTop: 45
  },
  ButtonHeader: {
    right: 16
  }
});
