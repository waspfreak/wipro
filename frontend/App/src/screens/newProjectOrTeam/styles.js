import { StyleSheet } from 'react-native';
export default StyleSheet.create({
  mainContent: {
    paddingVertical: 40,
    paddingHorizontal: 20
  }
  // header: {
  //   paddingHorizontal: 16
  // }
});
