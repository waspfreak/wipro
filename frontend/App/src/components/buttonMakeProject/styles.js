import { StyleSheet } from 'react-native';
import { colors, fonts } from '../../styles';

export default StyleSheet.create({
  container: {
    width: '90%',
    height: 201,
    paddingVertical: 12,
    paddingHorizontal: 16,
    borderColor: colors.border,
    borderWidth: 1,
    alignContent: 'center',
    alignSelf: 'center',
    alignItems: 'center',
    marginVertical: 5,
    textAlign: 'center'
  },
  iconContainer: {
    width: 40,
    height: 40,
    borderRadius: 99999,
    backgroundColor: colors.purple,
    alignItems: 'center',
    alignContent: 'center',
    marginTop: 14,
    marginBottom: 9,
    flexDirection: 'row',
    justifyContent: 'center'
  },
  title: {
    ...fonts.Title3Bold,
    marginBottom: 10
  },
  subtitle: {
    ...fonts.Caption0,
    textAlign: 'center',
    lineHeight: 18
  }
});
