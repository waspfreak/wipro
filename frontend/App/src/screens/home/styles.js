import { StyleSheet } from 'react-native';
import { colors, fonts } from '../../styles';
import globalStyles from '../../styles/globalStyles';

export default StyleSheet.create({
  container: {
    ...globalStyles.container,
    backgroundColor: colors.offWhite,
    paddingHorizontal: globalStyles.paddingHorizontal
  },
  topicsContainer: {
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
    alignContent: 'space-between',
    marginRight: -7
  },
  item: {
    width: '50%'
  },
  titleContainer: {
    paddingBottom: 0,
    paddingTop: 10
  },
  title: {
    ...fonts.Title2Bold
  },
  scrollBarContainer: {
    flex: 1,
    marginBottom: 30
  },
  searchContainer: {
    marginTop: 50,
    width: '100%',
    flex: 1
  },
  inputStyle: {
    backgroundColor: colors.inputBackground,
    borderRadius: 8,
    borderWidth: 0,
    height: 50,
    width: '100%',
    color: colors.darkGrey
  },
  headerWrapperStyle: {
    backgroundColor: 'transparent',
    borderBottomWidth: 0
  },
  imageContainer: {
    height: '100%',
    width: '50%',
    justifyContent: 'center',
    alignItems: 'center'
  },
  scrollBar: {
    marginTop: 10
  }
});
