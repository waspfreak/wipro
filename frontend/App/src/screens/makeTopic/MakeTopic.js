import React, { useState, useEffect } from 'react';
import { View } from 'react-native';
import { Header } from '../../components/header/Header';
import ButtonIcon from '../../components/buttonIcon/ButtonIcon';
import Button from '../../components/button/Button';
import InputField from '../../components/inputField/InputField';
import { SafeAreaView } from 'react-navigation';
import styles from './styles';
import { colors } from '../../styles';
import { oneTaskStoreInitialState } from '../../stores/initialGlobalState';
import { useGlobalStore } from '../../stores/globalStore';
import { testIds } from '../../constants/testIds';

export const MakeTopic = ({ navigation }) => {
  const [oneTaskStore, dispatch] = useGlobalStore();
  const [newTopic, setNewTopic] = useState({
    ...oneTaskStoreInitialState.topics.newTopic
  });

  useEffect(() => {
    if (newTopic.name.length > 0) {
      navigation.navigate('Topic');
    }
    // eslint-disable-next-line
  }, [oneTaskStore]);

  const leftMenu = () => {
    return (
      <ButtonIcon
        onPress={() => navigation.goBack()}
        color="transparent"
        icon="LeftChevron"
        iconColor={colors.blue}
        iconSizeWidth="28"
        iconSizeHeight="28"
        marginLeft={16}
        testIdentifier={testIds.backButton}
      />
    );
  };
  const RightMenu = () => {
    return (
      <Button
        wrapperStyle={styles.ButtonHeader}
        onPress={() => dispatch({ type: 'NEW_TOPIC', payload: { newTopic } })}
        title="Save"
        small
        tertiary
        disabled={newTopic.name === ''}
        testIdentifier={testIds.saveButton}
      />
    );
  };

  return (
    <SafeAreaView style={styles.container}>
      <Header
        wrapperStyle={styles.wrapperStyle}
        title="Make a Topic"
        left={leftMenu()}
        right={RightMenu()}
      />

      <View style={styles.mainContent}>
        <InputField
          label="Name this topic"
          placeholder="e.g. Office Renovation"
          onChangeText={name => setNewTopic({ ...newTopic, name })}
        />
        <InputField
          simple
          multiline
          inputStyle={styles.input}
          label="Add an additional description"
          placeholder="e.g. Plans and scheduling for expanding the office"
          onChangeText={description =>
            setNewTopic({ ...newTopic, description })
          }
        />
      </View>
    </SafeAreaView>
  );
};

export default MakeTopic;

MakeTopic.navigationOptions = {
  header: null
};
