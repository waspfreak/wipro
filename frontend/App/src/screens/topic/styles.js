import { StyleSheet } from 'react-native';
import { colors, fonts } from '../../styles';

export default StyleSheet.create({
  mainContent: {
    paddingVertical: 20
  },
  topicContainer: {
    paddingVertical: 16,
    paddingHorizontal: 24,
    borderBottomColor: colors.border,
    borderBottomWidth: 1
  },
  title: {
    ...fonts.Title2Bold,
    paddingBottom: 6
  },
  description: {
    ...fonts.Headline
  },
  infoContainer: {
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
    paddingTop: 20,
    borderBottomColor: colors.border,
    borderBottomWidth: 1
  },
  activityContainer: {
    paddingVertical: 12,
    paddingHorizontal: 24
  },
  user: {
    paddingTop: 15
  },
  interfaceStyle: {
    width: 100,
    marginHorizontal: 10,
    marginBottom: 24
  }
});
