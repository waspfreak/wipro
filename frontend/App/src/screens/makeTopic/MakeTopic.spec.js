import React from 'react';
import {
  render,
  fireEvent,
  waitForElement
} from '@testing-library/react-native';
import MakeTopic from './MakeTopic';
import { navigationPropMock } from '../../utils/propMocks';
import { StateProvider } from '../../stores/globalStore';
import { GlobalReducer } from '../../stores/globalReducer';
import { oneTaskStoreInitialState } from '../../stores/initialGlobalState';
import styles from '../../components/button/styles';
import { testIds } from '../../constants/testIds';

describe('Create Topic Screen tests', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      navigation: navigationPropMock
    };

    component = render(
      <StateProvider
        initialState={oneTaskStoreInitialState}
        reducer={GlobalReducer}
      >
        <MakeTopic {...props} />
      </StateProvider>
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match Topic snapshot', () => {
    const { baseElement } = component;
    expect(baseElement).toMatchSnapshot();
  });

  it('should call navigate back when back button is pressed', () => {
    const { getByTestId } = component;
    const backChevron = getByTestId(testIds.backButton);
    fireEvent.press(backChevron);
    expect(navigationPropMock.goBack).toHaveBeenCalled();
  });

  it('should dispatch new topic event when save is pressed', async () => {
    const { getByPlaceholderText, getByTestId } = component;
    const [
      nameTextInput,
      descriptionInput,
      saveButton
    ] = await waitForElement(() => [
      getByPlaceholderText('e.g. Office Renovation'),
      getByPlaceholderText(
        'e.g. Plans and scheduling for expanding the office'
      ),
      getByTestId(testIds.saveButton)
    ]);
    fireEvent.changeText(nameTextInput, 'My topic name');
    fireEvent.changeText(descriptionInput, 'My topic name');
    fireEvent.press(saveButton);
    expect(navigationPropMock.navigate).toBeCalledWith('Topic');
  });

  it('should enable menu button when user enters a topic', async () => {
    const { getByPlaceholderText, getByTestId } = component;
    const [nameTextInput, saveButton] = await waitForElement(() => [
      getByPlaceholderText('e.g. Office Renovation'),
      getByTestId(testIds.saveButton)
    ]);
    expect(saveButton.props.style[0].backgroundColor).toBe(
      styles.containerDisabled.backgroundColor
    );
    fireEvent.changeText(nameTextInput, 'My topic name');
    expect(saveButton.props.style[0].backgroundColor).toBe(
      styles.containerTertiary.backgroundColor
    );
  });
});
