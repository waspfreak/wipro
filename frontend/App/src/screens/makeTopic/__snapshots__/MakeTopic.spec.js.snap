// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Topic Screen tests should match Topic snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 20,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderBottomColor": "#c6c6c6",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "height": 50,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "left": 0,
              "position": "absolute",
            }
          }
        >
          <TouchableHighlight
            activeOpacity={0.85}
            delayPressOut={100}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 50,
                  "height": 28,
                  "justifyContent": "center",
                  "width": 28,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "marginLeft": 16,
                },
                undefined,
              ]
            }
            testID="backButton"
            underlayColor="#3478f6"
          >
            <View>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="28"
                bbWidth="28"
                focusable={false}
                height="28"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    undefined,
                    Object {
                      "opacity": 1,
                    },
                    Object {
                      "flex": 0,
                      "height": 28,
                      "width": 28,
                    },
                  ]
                }
                vbHeight={28}
                vbWidth={28}
                width="28"
              >
                <RNSVGGroup
                  fill={
                    Array [
                      0,
                      4278190080,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={Array []}
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                >
                  <RNSVGPath
                    d="M8.292 15.213a1.016 1.016 0 0 1 0-1.426l9.245-9.36a1.431 1.431 0 0 1 2.04 0c.563.57.563 1.496 0 2.066L11.67 14.5l7.909 8.005a1.476 1.476 0 0 1 0 2.066c-.564.57-1.479.57-2.04 0l-9.246-9.36z"
                    fill={
                      Array [
                        0,
                        4279986889,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={0}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={
                      Array [
                        "fill",
                        "fillRule",
                      ]
                    }
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    vectorEffect={0}
                    x={0}
                    y={0}
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </TouchableHighlight>
        </View>
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": 17,
            }
          }
        >
          Make a Topic
        </Text>
        <View
          style={
            Object {
              "position": "absolute",
              "right": 0,
            }
          }
        >
          <TouchableHighlight
            activeOpacity={0.85}
            delayPressOut={100}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#e5e5e5",
                  "borderColor": "#1b6ac9",
                  "borderRadius": 50,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "minHeight": 30,
                  "minWidth": 63,
                  "padding": 5,
                  "paddingHorizontal": 16,
                },
                Object {
                  "right": 16,
                },
              ]
            }
            testID="saveButton"
            underlayColor="black"
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                  }
                }
              >
                Save
              </Text>
            </View>
          </TouchableHighlight>
        </View>
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 20,
            "paddingVertical": 20,
          }
        }
      >
        <View
          style={
            Object {
              "paddingBottom": 10,
              "paddingTop": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "SFProDisplay-Bold",
                "fontSize": 17,
                "marginBottom": 11,
                "paddingBottom": 11,
              }
            }
          >
            Name this topic
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingTop": 10,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              placeholder="e.g. Office Renovation"
              placeholderTextColor="rgba(60, 60, 67, 0.6)"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "borderColor": "rgba(60, 60, 67, 0.6)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "color": "#000000",
                    "fontFamily": "SFProDisplay-Regular",
                    "fontSize": 17,
                    "height": 44,
                    "padding": 10,
                    "paddingTop": 10,
                    "width": "100%",
                    "zIndex": 0,
                  },
                  undefined,
                  null,
                ]
              }
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
        <View
          style={
            Object {
              "paddingBottom": 10,
              "paddingTop": 20,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "SFProDisplay-Bold",
                "fontSize": 17,
                "marginBottom": 11,
                "paddingBottom": 11,
              }
            }
          >
            Add an additional description
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingBottom": 10,
                "paddingTop": 10,
              }
            }
          >
            <TextInput
              allowFontScaling={true}
              multiline={true}
              placeholder="e.g. Plans and scheduling for expanding the office"
              placeholderTextColor="rgba(60, 60, 67, 0.6)"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "borderColor": "rgba(60, 60, 67, 0.6)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "color": "#000000",
                    "fontFamily": "SFProDisplay-Regular",
                    "fontSize": 17,
                    "height": 44,
                    "padding": 10,
                    "paddingTop": 10,
                    "width": "100%",
                    "zIndex": 0,
                  },
                  Object {
                    "height": 80,
                    "marginTop": 45,
                    "width": "100%",
                  },
                  null,
                ]
              }
              underlineColorAndroid="transparent"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
